code/obj/item/ai_modules.dm

```diff
@@ -411,7 +411,72 @@ AI MODULES
 		. = ..()
 		page_departments["Silicon"] = MGO_SILICON
 
+/obj/item/aiModule/hologram_expansion
+	name = "Hologram Expansion Module"
+	desc = "A module that updates an AI's hologram images."
+	var/expansion
 
+	install(obj/machinery/computer/aiupload/comp)
+		if (comp.status & NOPOWER)
+			boutput(usr, "\The [comp] has no power!")
+			return
+		if (comp.status & BROKEN)
+			boutput(usr, "\The [comp] computer is broken!")
+			return
+
+		src.transmitInstructions(usr)
+
+	transmitInstructions(var/mob/sender, var/law)
+		// what if we let them pick what AI to update?
+		// the future is now
+		// this is mostly stolen from observer.dm's observe list
+		var/list/names = list()
+		var/list/namecounts = list()
+		var/list/ais = list()
+		for_by_tcl(AI, /mob/living/silicon/ai)
+			LAGCHECK(LAG_LOW)
+			var/name = AI.name
+			if (name in names)
+				namecounts[name]++
+				name = "[name] ([namecounts[name]])"
+			else
+				names.Add(name)
+				namecounts[name] = 1
+
+			ais[name] = AI
+
+		var/mob/living/silicon/ai/AI = null
+		if (ais.len == 1)
+			AI = ais[names[1]]
+		else if (ais.len > 1)
+			var/res = input("Which AI are you modifying?", "Hologram update", null, null) as null|anything in ais
+			AI = ais[res]
+		else if (ais.len == 0)
```

Comment:
length

Reply:
Updated this and source of bad copypasta.